packages/cli/generators/relation/has-one-relation.generator.js

```diff
@@ -0,0 +1,165 @@
+
+module.exports = class HasOneRelationGenerator extends BaseRelationGenerator {
+  constructor(args, opts) {
+    super(args, opts);
+  }
+
+  async generateControllers(options) {
+    this.artifactInfo.sourceModelPrimaryKey = options.sourceModelPrimaryKey;
+    this.artifactInfo.sourceModelPrimaryKeyType =
+      options.sourceModelPrimaryKeyType;
+    this.artifactInfo.sourceModelClassName = options.sourceModel;
+    this.artifactInfo.targetModelClassName = options.destinationModel;
+    this.artifactInfo.paramTargetModel = utils.camelCase(
+      options.destinationModel,
+    );
+    this.artifactInfo.sourceRepositoryClassName =
+      this.artifactInfo.sourceModelClassName + 'Repository';
+    this.artifactInfo.controllerClassName =
+      this.artifactInfo.sourceModelClassName +
+      this.artifactInfo.targetModelClassName +
+      'Controller';
+
+    this.artifactInfo.paramSourceRepository = utils.camelCase(
+      this.artifactInfo.sourceModelClassName + 'Repository',
+    );
+
+    this.artifactInfo.sourceModelName = utils.toFileName(options.sourceModel);
+    this.artifactInfo.sourceModelPath = utils.pluralize(
+      this.artifactInfo.sourceModelName,
+    );
+    this.artifactInfo.targetModelName = utils.toFileName(
+      options.destinationModel,
+    );
+
+    this.artifactInfo.relationPropertyName = options.relationName;
+    this.artifactInfo.targetModelPrimaryKey =
+      options.destinationModelPrimaryKey;
+    this.artifactInfo.targetModelPrimaryKeyType =
+      options.destinationModelPrimaryKeyType;
+
+    const source = this.templatePath(CONTROLLER_TEMPLATE_PATH_HAS_ONE);
+
+    this.artifactInfo.name =
+      options.sourceModel + '-' + options.destinationModel;
+    this.artifactInfo.outFile =
+      utils.toFileName(this.artifactInfo.name) + '.controller.ts';
+
+    const dest = this.destinationPath(
+      path.join(this.artifactInfo.outDir, this.artifactInfo.outFile),
+    );
+
+    this.copyTemplatedFiles(source, dest, this.artifactInfo);
+    await relationUtils.addExportController(
+      this,
+      path.resolve(this.artifactInfo.outDir, 'index.ts'),
+      this.artifactInfo.controllerClassName,
+      utils.toFileName(this.artifactInfo.name) + '.controller',
+    );
+  }
+
+  async generateModels(options) {
+    const modelDir = this.artifactInfo.modelDir;
+    const sourceModel = options.sourceModel;
+
+    const targetModel = options.destinationModel;
+    const relationType = options.relationType;
+    const relationName = options.relationName;
+    const fktype = options.destinationModelPrimaryKeyType;
+
+    const project = new relationUtils.AstLoopBackProject();
+    const sourceFile = relationUtils.addFileToProject(
+      project,
+      modelDir,
+      sourceModel,
+    );
+    const sourceClass = relationUtils.getClassObj(sourceFile, sourceModel);
+
+    relationUtils.doesRelationExist(sourceClass, relationName);
+
+    const modelProperty = this.getHasOne(targetModel, relationName, fktype);
```

Comment:
ditto, `const modelProperty` should be similar to the hasMany generator.
And check `isForeignKeyExist`.